src/languages/php.js

```diff
@@ -379,12 +434,25 @@ export default function(hljs) {
       },
       VARIABLE,
       FUNCTION_INVOKE,
+      LEFT_AND_RIGHT_SIDE_OF_DOUBLE_COLON,
+      {
+        match: [
+          /const/,
+          /\s/,
+          IDENT_RE,
+          /\s*=/,
+        ],
+        scope: {
+          1: "keyword",
+          3: "variable.constant",
+        },
+      },
       {
         // swallow composed identifiers to avoid parsing them as keywords
         match: regex.concat(
-          /(::|->)+[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*/,
-          regex.concat("(?!", WHITESPACE, "*\\()"),
-          /(?![a-zA-Z0-9_\x7f-\xff])/
+          /(::|->)+/,
+          IDENT_RE,
+          regex.concat("(?!", WHITESPACE, "*\\()")
```

Comment:
Was there an issue with line 387?

Reply:
We define again the same regex pattern instead of using an already defined one.

Reply:
You changed the regex pattern... did you not?  It was previously ~:

-> ident [ NOT `\s(` AND NOT IDENT ]

You removed the "AND NOT IDENT", portion did you not? Was there a reason?

Reply:
Hmm, lines 385 and 387 are the same as the regex rule of IDENT_RE. IDENT_RE only has the suffix `(?![$]))`, which was missing here (in my opinion). [Documentation](https://www.php.net/manual/en/language.variables.basics.php)  states that all labels follow the same pattern:

> Variable names follow the same rules as other labels in PHP. A valid variable name starts with a letter or underscore, followed by any number of letters, numbers, or underscores. As a regular expression, it would be expressed thus: ^[a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]*$

I also changed the position of the negative look-ahead from the middle to the end. It didn't break any test and according to my knowledge about regex, it should work the same. If you could provide me with a sample that passes the old code and fail in the new one, then I would fix it.

Reply:
> It didn't break any test...

Sadly our tests are far from comprehensive, hence the careful eye I have with reviewing PRs.

This was added with: d49e9fb8fa82a43ed96333510c9c5bc1af776352

The test code added was `, 3 => \Location\Web\URI::class));` so I imagine this was to protect the `::class`... but now you're explicitly handling that... so if we remove this guard entirely, does it break anything?

Reply:
I see. Reverting back [to](https://github.com/highlightjs/highlight.js/commit/d49e9fb8fa82a43ed96333510c9c5bc1af776352#diff-b056c5a26741799f5d40c82d9bfff3d9975a8b63bdb5955cf5e30b3059042bddL76), didn't break anything.